### PR TITLE
set `PFun` free of `PLift`. Rebuilt entirely with Scott Plut

### PR DESCRIPTION
**Pull request by SeungheonOh**

### CODE REVIEW

File: src/Plutarch/Test/QuickCheck/Function.hs [hunk 1]
```diff
@@ -144,16 +136,20 @@ showPFun (PFun t d _) =
             )
         ++ "\n}"
 
+conTable :: [(TestableTerm a, TestableTerm b)] -> TestableTerm (PList (PPair a b))
```
Comment (SeungheonOh) [inline src/Plutarch/Test/QuickCheck/Function.hs]: `forall`

**Review by kozross — APPROVED** [review]

**Comment by SeungheonOh** [comment]
Turns out generating Hash takes longer than compiling it.

**Comment by SeungheonOh** [comment]
I ran some more test and turns out this implementation is quite a bit slower than Lifty one, naturally. Knowing that, I don't want to fully deprecate the Lifty PFun. 

I'm currently struggling to fine a good way to combine them together...

**Comment by SeungheonOh** [comment]
I tried putting `PFunLift` as a second constructor in `PFun`, and it doesn't work because it causes very complex mess up in Constraints. 

Second best solution is making a separate type `PFunLift`, but this would make `fromPFun` less terse because it will require user to specify which `PFun` they want to use...

**Pull request closed without merging**